app/javascript/packs/pw-strength.js

```diff
@@ -14,6 +14,8 @@ const scale = {
   4: ['pw-great', I18n.t('instructions.password.strength.v')],
 };
 
+const snakeCase = (string) => string.replace(/[ -]/g, '_').replace(/\W/g, '').toLowerCase();
```

Comment:
should we add an `inflector.js` and share this function across files? Or put it in the i18n module somewhere?

Reply:
> should we add an `inflector.js` and share this function across files? Or put it in the i18n module somewhere?

I had it in my mind to want to move away from needing these at all, toward either (a) hard-coding the specific strings we're using and/or (b) changing the sets of the interpolated keys to ones which are already properly-formatted.

Though to my other comment, maybe we just pull in Lodash for [its implementation](https://lodash.com/docs/4.17.15#snakeCase).

Reply:
Seems like it would be useful enough, I forget if we have tree-shaking compilers, because if so that would make it a no-brainer to include

Reply:
It would be tree-shaken, and further, Webpack chunking would handle optimizing its usage across packages if we started using it elsewhere as well. The main downsides for me would be:

- Tree-shaken Lodash still tends to produce larger output than you might expect or hope for
- Benefits of chunking are lost if-ever we publish packages standalone
- Depending how widespread we choose to use Lodash, knowledge of its API could be seen as a barrier to entry

None of these are really deal-breakers for me. On the up-side, we also have a fair few things which are essentially reimplementations of what Lodash could already offer us, e.g.:

- [`useInstanceId`](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/document-capture/hooks/use-instance-id.js) vs. [`_.uniqueId`](https://lodash.com/docs/4.17.15#uniqueId)
- [`except`](https://github.com/18F/identity-idp/blob/bdd15a68d69f33d42de8128d92cc5dbd9466b860/app/javascript/packages/document-capture/components/document-capture.jsx#L23-L40) vs. [`_.omit`](https://lodash.com/docs/4.17.15#omit)
- [`kebabCase`](https://github.com/18F/identity-idp/blob/bdd15a68d69f33d42de8128d92cc5dbd9466b860/app/javascript/packages/page-data/index.js#L1-L15) vs. [`_.kebabCase`](https://lodash.com/docs/4.17.15#kebabCase)

Reply:
maybe we just take our three functions and put them in a file and call it `logindash` ;)

I don't feel that strongly either way

Reply:
According to [Bundlephobia](https://bundlephobia.com/package/lodash-es@4.17.21), 0.8kb and 2.3kb for `escapeRegExp` and `snakeCase` respectively, possibly with some overlap between.

Reply:
I think I'll plan to leave this as-is for now, to the initial point of wanting to move away from needing to convert the keys, and the [other conversation](https://github.com/18F/identity-idp/pull/5459#discussion_r719667126) finding an alternative implementation which doesn't require escaping regular expressions.